identity/src/main/java/com/android/identity/L2CAPServer.java

```diff
@@ -47,84 +47,99 @@ class L2CAPServer {
     private BluetoothServerSocket mServerSocket;
     private BluetoothSocket mSocket;
     private BlockingQueue<byte[]> mWriterQueue = new LinkedTransferQueue<>();
-    private BlockingQueue<byte[]> mMessageReceivedQueue = new LinkedTransferQueue<>();
     Thread mWritingThread;
-    Thread mReportReceivedMessageThread;
 
     L2CAPServer(@Nullable L2CAPServer.Listener listener, @LoggingFlag int loggingFlags) {
         mListener = listener;
         mLog = new Util.Logger(TAG, loggingFlags);
     }
 
     @RequiresApi(api = Build.VERSION_CODES.Q)
-    boolean start(@NonNull BluetoothAdapter bluetoothAdapter) {
+    byte[] start(@NonNull BluetoothAdapter bluetoothAdapter) {
         try {
             // Using insecure L2CAP allows the app to use L2CAP friction less, otherwise
             // Android will require bluetooth pairing with other device showing the pairing code
             mServerSocket = bluetoothAdapter.listenUsingInsecureL2capChannel();
+
+            Thread waitingForConnectionThread = new Thread(this::waitForConnectionThread);
+            waitingForConnectionThread.start();
+
+            int psm = mServerSocket.getPsm();
+            byte[] psmValue = ByteBuffer.allocate(4).putInt(psm).array();
```

Comment:
Is this guaranteed to always give us the correct endianness?

Reply:
I think so. It's actually not clear to me how this should be encoded, right now 18013-5 Appendix A just says "The contents of the characteristics shall be the Protocol Service Multiplexer (PSM)." but looking through the BT specs it's not clear what the encoding is in this case. I have a question out to the editor for a possible clarification. I'll add a TODO here so we won't forget to revisit.